src/plugins/ResizeReorder/DragProxy.js

```diff
@@ -0,0 +1,286 @@
+/**
+ * Copyright Schrodinger, LLC
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ReorderHandle
+ * @typechecks
+ * @file Contains the reordering logic
+ */
+
+import React from 'react';
+import cx from '../../vendor_upstream/stubs/cx';
+import PropTypes from 'prop-types';
+import DOMMouseMoveTracker from '../../vendor_upstream/dom/DOMMouseMoveTracker';
+import { FixedDataTableContext } from '../../FixedDataTableContext';
+import requestAnimationFramePolyfill from '../../vendor_upstream/core/requestAnimationFramePolyfill';
+import cancelAnimationFramePolyfill from '../../vendor_upstream/core/cancelAnimationFramePolyfill';
+import _ from 'lodash';
+
+const DRAG_SCROLL_SPEED = 15;
+const DRAG_SCROLL_BUFFER = 100;
+
+class DragProxy extends React.PureComponent {
```

Comment:
The drag proxy is a react component used by ReorderCell that takes care of dragging the specified node along the mouse cursor.

It works by injecting a copy of the original node into a place holder div, and the placeholder gets translated with the users's cursor.
DragProxy also consumes `FixedDataTableContext` to figure out stuff like table bounds, target drop position, dispatching actions to scroll the table, etc.

The best order to read this file would be 
1. render
2. componentDidMount
3. onMouseDown/startDrag
4. onMouseMove/updateDisplacementWithScroll/getBoundedDeltaX
5. onMouseUp/updateColumnOrder